drivers/can/Kconfig.nxp_s32

```
@@ -1,10 +1,11 @@
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 config CAN_NXP_S32_CANXL
 	bool "NXP S32 CANXL driver"
 	default y
 	depends on DT_HAS_NXP_S32_CANXL_ENABLED
+	depends on CLOCK_CONTROL
```

Should we select clock control instead?

```suggestion
	select CLOCK_CONTROL
```

---

agree, I was hesitant on `depends on` or `select`, but after re-reading the purpose of each, it makes more sense to use `select`.